test/unit/mesh-gateway-deployment.bats

```diff
@@ -0,0 +1,505 @@
+  [ $(echo "$actual" | yq -r '.readinessProbe.tcpSocket.port')  = "8443" ]
+}
+
+@test "meshGateway/Deployment: wanAddress.port defaults to 443" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'client.grpc=true' \
+      --set 'meshGateway.wanAddress.useNodeIP=true' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].command[2]' | tee /dev/stderr)
+  [[ "${actual}" =~ '-wan-address="${HOST_IP}:443"' ]]
+}
+
+@test "meshGateway/Deployment: wanAddress uses NodeIP by default" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'client.grpc=true' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].command[2]' | tee /dev/stderr)
+  [[ "${actual}" =~ '-wan-address="${HOST_IP}:443"' ]]
+}
+
+@test "meshGateway/Deployment: wanAddress.useNodeIP" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'client.grpc=true' \
+      --set 'meshGateway.wanAddress.useNodeIP=true' \
+      --set 'meshGateway.wanAddress.port=4444' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].command[2]' | tee /dev/stderr)
+  [[ "${actual}" =~ '-wan-address="${HOST_IP}:4444"' ]]
+}
+
+@test "meshGateway/Deployment: wanAddress.useNodeName" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'client.grpc=true' \
+      --set 'meshGateway.wanAddress.useNodeIP=false' \
+      --set 'meshGateway.wanAddress.useNodeName=true' \
+      --set 'meshGateway.wanAddress.port=4444' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].command[2]' | tee /dev/stderr)
+  [[ "${actual}" =~ '-wan-address="${NODE_NAME}:4444"' ]]
+}
+
+@test "meshGateway/Deployment: wanAddress.host" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'client.grpc=true' \
+      --set 'meshGateway.wanAddress.useNodeIP=false' \
+      --set 'meshGateway.wanAddress.useNodeName=false' \
+      --set 'meshGateway.wanAddress.host=myhost' \
+      --set 'meshGateway.wanAddress.port=4444' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].command[2]' | tee /dev/stderr)
+  [[ "${actual}" =~ '-wan-address="myhost:4444"' ]]
+}
+
+@test "meshGateway/Deployment: fails if consulServiceName is set and bootstrapACLs is true" {
```

Comment:
As mentioned elsewhere, it would be nice for this to work if someone happened to set the service name to `mesh-gateway`. A test for that case would be good.

Reply:
✅